druid/src/modal.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::any::Any;
+
+use crate::lens;
+use crate::widget::prelude::*;
+use crate::{Data, Point, Selector, SingleUse, WidgetExt, WidgetPod};
+
+/// Describes a modal widget.
+///
+/// A modal widget is a widget that can be displayed over all the other widgets in a window. It
+/// consists of a widget (which must take the same data type as the [`Window`]) and some settings
+/// describing how the widget will be presented.
+///
+/// You can display a modal widget by sending a [`SHOW_MODAL`] command to a window.
+///
+/// [`Window`]: struct.Window.html
+/// [`SHOW_MODAL`]: struct.Modal.html#associatedconstant.SHOW_MODAL
+pub struct ModalDesc<T> {
+    widget: Box<dyn Widget<T>>,
+    /// If false, only the modal will get user input events.
+    pass_through_events: bool,
+    /// If set, the origin of the modal widget. If unset, the modal widget is centered in the
+    /// `ModalHost`.
+    position: Option<Point>,
```

Comment:
This might be better expressed as a UnitPoint? Difficult to have a meaningful number for anything other than the top left of the screen. I was playing around with using this to create a "toast" notification:
```rust
    ModalDesc::new(
        Flex::row()
            .with_child(toast_label)
            .with_child(close_button)
            .padding(20.0)
            .background(COLOR_VERY_LIGHT)
            .rounded(4.)
            .align_vertical(UnitPoint::TOP)
            .align_right()
            .expand()
            .padding(10.),
    )
    .position(Point::new(0., 0.)
```

Reply:
It definitely makes sense to have more positioning flexibility, but there's actually a use-case for `Point`: tooltips, which you want to display near the mouse position, for which it's easy to get window-relative coordinates.

Reply:
Ah, that makes sense.